3]. Competitive Programming/LeetCode/NQueens_51.java

```diff
@@ -0,0 +1,74 @@
+
+class NQueens_51 {
+    public static List<List<String>> solveNQueens(int n) {
+        // create n x n Board
+        int[][] board = new int[n][n];
+        List<List<String>> result = new ArrayList<>();
+        solveNQueensHelper(board, 0, new ArrayList<String>(), result);
+        return result;
```

Comment:
Error
```suggestion
        return result;
```

Reply:
Code is error free. Infact, this is the solution that I was able to submit on Leetcode successfully.
Could you please elaborate on the error that you got when you tried this solution ?

Reply:
![image](https://user-images.githubusercontent.com/60146338/122214399-ab23dc80-cec7-11eb-9955-a5d30f25cc40.png)

check this

Reply:
![image](https://user-images.githubusercontent.com/60146338/122214769-1bcaf900-cec8-11eb-98dd-991860d26dbd.png)

Change the class name in code of 1st line

And add the solution in folder ( `Problems` )
code name like `_0051)_N-Queens.java`

Reply:
Yes that makes absolute sense, solution provided in the file has been named as NQueens_51.
All the solutions on leetcode should be present in the Solution class, when you tried to run execute the solution, you have copied the entire content of the file.
Only copy the methods inside the provided class, and make sure that you do not copy the main method.

Sample Reference: 
![Accepted Solution](https://user-images.githubusercontent.com/6636804/122217039-a9a7e380-ceca-11eb-9fa3-23dd0d37c2ef.png)